common/numerated/src/tree.rs

```diff
@@ -203,6 +200,7 @@ impl<T: Numerated> IntervalsTree<T> {
 
         if let Some(right_end) = right_end.inc_if_lt(start) {
             if right_end == start {
+                debug_assert!(right_start <= end, "Must be cause of method it was found");
                 self.inner.insert(right_start, end);
             } else {
```

Comment:
btw couldn't remove/insert fns be optimized with bin search mb ?

Reply:
case of tree like 1, 3, 5, 7, 9... etc

Reply:
it's already optimised with bin search. see BTreeMap::range